### DISCUSSION TITLE
So...

### DISCUSSION THREAD

**Discussion by rockumk** [jpcima/ysfx#49] [General]
Sorry to ask such a basic question, but I am trying to make sure... the purpose of this is to essentially wrap a JSFX plugin inside of a VST3 and/or AU unit?  Does it result in a distributable plugin?

**Comment by frank1119** [comment]
This plugin makes it possible to execute jsxf files. Jsxf is a language for processing midi-data and/or audio-samples. Originally it was embedded in Reaper. Later on Cockos created a vst for running this jsxf programs in other environments, but they are not paying too much attention to this plugin anymore, so this project goes on where Cockos stopped.

It is distributable in this way that you can give the jsfx file and the ysfx plugin to somebody else. It does not create a separate .vst3 plugin with the jsfx code embedded or something like that. The plugins (jsxf files) you create are fully disclosed to the people you distribute them to. 

See [Reaper JSFX](https://www.reaper.fm/sdk/js/js.php) for more information on jsfx programming (if you should need that)

**Reply by rockumk** [reply]
Ok thanks.  I've only created one JSFX plug and it was pretty Reaper specific.  I applaud your efforts.  There are some very nice JSFX plugins.  Kind of you to make those more readily available outside the Reaper community.

**Comment by jpcima** [comment]
> Sorry to ask such a basic question, but I am trying to make sure... the purpose of this is to essentially wrap a JSFX plugin inside of a VST3 and/or AU unit? Does it result in a distributable plugin?

You are correct that this software has a wrapper of JSFX into an audio plugin.
That's not the whole of it however.

It has a library that software programmers can use to add JSFX support to their software.
One could use that to change a JSFX into a redistributable plugin.
I didn't provide a template though, that would facilitate that possibility.

There exist other perspectives, such as DAWs, game engines.
A first goal for is to get JSFX implemented into the Carla software (and by extension Zrythm which is reliant on the same code).
